From bbecd0e5f9da64f907d71bbdf78094dfbcf56fba Mon Sep 17 00:00:00 2001
From: Tristan Cacqueray <tdecacqu@redhat.com>
Date: Wed, 7 Dec 2016 07:30:27 +0000
Subject: [PATCH] Ignore bad requests in notification hook

The notification engine is raising internal server error when trying
to process invalid requests such as unauthorized call. This change
avoid processing request with status code >= 400.

Change-Id: Id7b9f4b342eebcc6258ecc6786e7c0b81c77def4
---
 storyboard/notifications/notification_hook.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/storyboard/notifications/notification_hook.py b/storyboard/notifications/notification_hook.py
index 1058475..aab541c 100644
--- a/storyboard/notifications/notification_hook.py
+++ b/storyboard/notifications/notification_hook.py
@@ -68,6 +68,10 @@ class NotificationHook(hooks.PecanHook):
         if state.request.method not in ['POST', 'PUT', 'DELETE']:
             return
 
+        # Ignore requests that failed
+        if state.response.status_code >= 400:
+            return
+
         request = state.request
         response = state.response
 
-- 
2.7.3

